controllers/azurecluster_controller.go

```diff
@@ -115,6 +115,8 @@ func (acr *AzureClusterReconciler) SetupWithManager(ctx context.Context, mgr ctr
 // +kubebuilder:rbac:groups=resources.azure.com,resources=resourcegroups/status,verbs=get;list;watch
 // +kubebuilder:rbac:groups=network.azure.com,resources=natgateways,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=network.azure.com,resources=natgateways/status,verbs=get;list;watch
+// +kubebuilder:rbac:groups=network.azure.com,resources=bastionhosts,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=network.azure.com,resources=bastionhosts/status,verbs=get;list;watch
```

Comment:
You could also club these annotations with the natgateways annotations defined above. 
Shall we be discretely listing the kubebuilder annotations per resource type (like it has been done here) or should we be clubbing it in one like for the same group type? (as demonstrated below)

I prefer the former existing implementation, pointing it out for discussion. 
```suggestion
// +kubebuilder:rbac:groups=network.azure.com,resources=natgateways;bastionhosts,verbs=get;list;watch;create;update;patch;delete
// +kubebuilder:rbac:groups=network.azure.com,resources=natgateways/status;bastionhosts/status,verbs=get;list;watch
```